pkgs/development/compilers/cudatoolkit/common.nix

```diff
@@ -199,6 +205,12 @@ stdenv.mkDerivation rec {
         patchelf --set-rpath "$rpath2" --force-rpath $i
       fi
     done < <(find $out $lib $doc -type f -print0)
+    ${lib.optionalString (lib.versionAtLeast version "11") ''
+      while IFS= read -r -d $'\0' i; do
+        echo "patching $i..."
+        patchelf --set-rpath "$rpath:$out/target-linux-x64" $i
```

Comment:
where is `$rpath` defined?

Reply:
also this seems to be adding `$out/target-linux-x64` to files in `$out/target-linux-x64`? if that's the case, we should just use `$ORIGIN` instead. See eg

https://github.com/NixOS/nixpkgs/blob/54da1bc6928c903956f807b653cd93d539a99002/pkgs/development/libraries/science/math/cudnn/generic.nix#L46

Reply:
> where is $rpath defined?

It's defined on [L64](https://github.com/mcwitt/nixpkgs/blob/37f2a7a020d40590b40a8fbd77f76cec839046ff/pkgs/development/compilers/cudatoolkit/common.nix#L64)

> we should just use $ORIGIN instead

Ah good call, thanks! I had even tried this originally but missed escaping the `$` so it didn't work. Fixed in 
2ceec4d.

Reply:
> It's defined on [L64](https://github.com/mcwitt/nixpkgs/blob/37f2a7a020d40590b40a8fbd77f76cec839046ff/pkgs/development/compilers/cudatoolkit/common.nix#L64)

Ah I see. Since `rpath` is only ever used in `preFixup`, let's move its definition into there. No need to pollute the larger `mkDerivation`. I know that you part of the code was pre-existing, but it's some low-hanging fruit while we're here.

Reply:
Good call; I didn't notice it was only used in `preFixup`. Moved the definition into a `let` in `preFixup` and also tried to clean up a bit (e.g. removing `rpath2`)